### PR TITLE
refactor: add callback info in status

### PR DESCRIPTION
**Pull request by shotasilagadzetaal**
## Description of Changes

So, with this PR when submitting a transaction we see the callback data in the response that was provided in the request.

When requesting transaction status we see all the callbacks added for the transaction. To add a new callback we can post the same transaction with a new callback that we want to add.
 
## Linked Issues / Tickets

Reference any related issues or tickets, e.g. "Closes #123".

## Testing Procedure

Describe the tests you've added or any testing steps you've taken.

- [x] I have added new unit tests
- [x] All tests pass locally
- [x] I have tested manually in my local environment

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have updated `CHANGELOG.md` with my changes

### CODE REVIEW

File: pkg/api/arc.yaml [hunk 1]
```diff
@@ -609,6 +627,11 @@ components:
           description: Transaction status
           example: "ACCEPTED_BY_NETWORK"
           nullable: false
+        callbacks:
```
Comment (boecklim) [inline pkg/api/arc.yaml:630]: I don't think it is needed to add this information to the api response

File: internal/metamorph/server.go [hunk 2]
```diff
@@ -244,6 +244,16 @@ func (s *Server) processTransaction(ctx context.Context, waitForStatus metamorph
...
 	}
+
+	for _, cb := range data.Callbacks {
+		if cb.CallbackURL != "" {
+			returnedStatus.Callbacks = append(returnedStatus.Callbacks, &metamorph_api.Callback{
+				CallbackUrl:   cb.CallbackURL,
+				CallbackToken: cb.CallbackToken,
+			})
+		}
+	}
```
Comment (boecklim) [inline internal/metamorph/server.go:255]: Is this just for completeness? Because I think that will have no effect

Comment (shotasilagadzetaal) [reply internal/metamorph/server.go:255]: For post request, I just include the callback data in the response. The data that was provided in the request.

File: pkg/api/arc.yaml (hunk 1 above)
Comment (shotasilagadzetaal) [reply pkg/api/arc.yaml:630]: It's just to show what callback was provided in the request.

Comment (boecklim) [reply pkg/api/arc.yaml:630]: I think we shouldn't add this field unless somebody asks for it.
Once something is in the api response, it is not simple to remove it again

**Comment by shotasilagadzetaal** [comment]
Getting status returns callbacks now. Post transaction response hasn't changed.

```
{
    "blockHash": "",
    "blockHeight": 0,
    "callbacks": [
        {
            "token": "token2",
            "url": "localhost:9298"
        }
    ],
    "competingTxs": null,
    "extraInfo": "",
    "merklePath": "",
    "timestamp": "2024-12-10T12:33:38.446713+04:00",
    "txStatus": "ACCEPTED_BY_NETWORK",
    "txid": "dc5a81817910124fc97811f9026e120515708b333bac596daccae7ea0e1d36c2"
}
```

File: internal/metamorph/server.go [hunk 3]
```diff
@@ -438,6 +448,17 @@ func (s *Server) GetTransactionStatus(ctx context.Context, req *metamorph_api.Tr
 		RejectReason: data.RejectReason,
 		CompetingTxs: data.CompetingTxs,
 		MerklePath:   data.MerklePath,
+		Timestamp:    timestamppb.New(data.StoredAt),
```
Comment (boecklim) [inline internal/metamorph/server.go:451]: It is not the `storedAt` field that we want to compare with. It is the field `LastSubmitted`

File: internal/metamorph/metamorph_api/metamorph_api.proto [hunk 4]
```diff
@@ -95,6 +101,8 @@ message TransactionStatus {
   uint64 block_height = 7;
   string block_hash = 8;
   string merkle_path = 9;
+  google.protobuf.Timestamp timestamp = 10;
```
Comment (boecklim) [inline internal/metamorph/metamorph_api/metamorph_api.proto:104]: Rename this to `last_submitted`

**Review by boecklim — APPROVED** [review]

**Pull request merged**